logstash-core/src/main/java/org/logstash/plugins/pipeline/AddressState.java

```diff
@@ -0,0 +1,75 @@
+package org.logstash.plugins.pipeline;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Class for representing the state of an internal address.
+ */
+public class AddressState {
+    private final String address;
+    private final Set<PipelineOutput> outputs = ConcurrentHashMap.newKeySet();
+    private volatile PipelineInput input = null;
```

Comment:
:tada: @andrewvc you're not leaving me much room for nitpicks. Excellent use of `volatile`.